src/server.c

```diff
@@ -6904,6 +6904,17 @@ int main(int argc, char **argv) {
                 if (argc_tmp == 1) {
                     /* Means that we only have one option name, like --port or "--port " */
                     handled_last_config_arg = 0;
+
+                    if ((j != argc-1) && argv[j+1][0] == '-' && argv[j+1][1] == '-' &&
+                        !strcasecmp(argv[j], "--save"))
+                    {
+                        /* Special case: handle some thing like `--save --config value`.
+                         * In this case, if next argument starts with `--`, we will reset
+                         * handled_last_config_arg flag and append an empty "" config value
+                         * to the options, so it will become `--save "" --config value`. */
```

Comment:
let's explain that we are doing it to be compatible with pre 7.0 behavior, since there might be users who generate a command line from an array and when it's empty that's what they produce.